### ISSUE TITLE
Button styles specified for different states are not applied

### ISSUE THREAD

**Issue by ezze** [GeekyAnts/NativeBase#4445]
**Describe the bug**

Button styles specified for `_hover` and `_pressed` states are not applied (it doesn't matter whether these styles are defined in a theme or inline).

**To Reproduce**

*src/theme.index.ts*

```typescript
import { extendTheme } from "native-base";

const theme = extendTheme({
  components: {
    Button: {
      variants: {
        solid: {
          _text: {
            color: "#fff"
          },
          backgroundColor: "#090",
          borderColor: "#060",
          _hover: {
            backgroundColor: "#b00",
            borderColor: "#900"
          },
          _pressed: {
            backgroundColor: "#333",
            borderColor: "#000"
          },
          _web: {
            _hover: {
              backgroundColor: "#b00",
              borderColor: "#900"
            },
            _pressed: {
              backgroundColor: "#333",
              borderColor: "#000"
            }
          }
        }
      }
    }
  }
});

export type ITheme = typeof theme;

export default theme;
```

*src/components/Example.tsx*

```typescript
import React from "react";
import { VStack, Button } from "native-base";

function Example() {
  return (
    <VStack space={2} width={500}>
      <Button>My styles from theme don't work</Button>
      <Button
        _hover={{
          backgroundColor: "#b00",
          borderColor: "#900"
        }}
        _pressed={{
          backgroundColor: "#333",
          borderColor: "#000"
        }}
      >
        My inline styles also don't work
      </Button>
    </VStack>
  );
}

export default Example;
```

**Expected behaviour**

`_hover` and `_pressed` styles are applied to button from both theme and inline.

**CodeSandBox/Snack link**
[Here is a live example](https://codesandbox.io/s/trusting-faraday-df7xd?file=/src/components/Example.tsx)

**Platform**

- NativeBase version 3.2.2
- Browser Chromium 
- Browser version 96.0.4664.110 (Official Build) snap (64-bit)
- React Native version 0.64.2
- node 14.15.1
- yarn 1.22.17

**Comment by rlingineni** [comment]
+1 on this. Hitting the same problem on Web. I cannot change the borderColor when focusing on a button

**Comment by rlingineni** [comment]
@surajahmed  - I saw that you closed #4428 , however, the expected behavior still seems broken. I'm unable to apply `borderColor` with the global theme or local focus settings.

https://codesandbox.io/s/trusting-faraday-df7xd?file=/src/components/Example.tsx

**Comment by vipulbhj** [comment]
@rlingineni 

<img width="1078" alt="Screenshot 2022-01-16 at 6 49 41 PM" src="https://user-images.githubusercontent.com/33618798/149661624-dfe17bd3-eef7-4dde-a3bc-c6d6e8763b7f.png">

Maybe It's because you haven't specified any `borderWidth`, I am also using `native-base@3.2.2` in my example, so you should also try version bumping 😄 

Do try and let us know, if it still doesn't work for you

**Comment by kevinvugts** [comment]
> @rlingineni
> 
> <img alt="Screenshot 2022-01-16 at 6 49 41 PM" width="1078" src="https://user-images.githubusercontent.com/33618798/149661624-dfe17bd3-eef7-4dde-a3bc-c6d6e8763b7f.png">
> 
> <img alt="Screenshot 2022-01-16 at 6 52 58 PM" width="1078" src="https://user-images.githubusercontent.com/33618798/149661752-77cd626d-091d-4e90-85da-e58bdc4bd222.png">
> 
> Maybe It's because you haven't specified any `borderWidth`, I am also using `native-base@3.2.2` in my example, so you should also try version bumping 😄
> 
> Do try and let us know, if it still doesn't work for you

I can confirm this doesn't work for the latest version of native-base (version 3.3.3)

**Comment by JaapWeijland** [comment]
I confirm! Cannot set the borderColor of an input.

```
<Input
      placeholder="E-mailadres"
      autoCapitalize="none"
      size="xl"
      px="4"
      py="3"
      autoCompleteType="email"
      onChangeText={handleChange('email')}
      onBlur={handleBlur('email')}
      value={values.email}
     _focus={{borderColor: 'red.500'}}
/>
```

This won't show a red border.

**Comment by devagero** [comment]
@ezze can you try with `bg` instead of `backgroundColor`? It works for me:

`
      <Button
        _hover={{
          bg: "#b00",
          borderColor: "#900"
        }}
        _pressed={{
          bg: "#333",
          borderColor: "#000"
        }}
      >
        My inline styles also don't work
      </Button>
`

**Comment by ankit-tailor** [comment]
Hi @JaapWeijland, this issue is fixed and will be released in the upcoming version.

**Comment by surajahmed** [comment]
Thank you so much for your patience. This issue has been fixed in version `3.3.5` or later.
